nautobot/dcim/views.py

```diff
@@ -273,6 +276,105 @@ class LocationTypeUIViewSet(NautobotUIViewSet):
 #
 
 
+class RackGroupsFieldsPanel(object_detail.ObjectFieldsPanel):
+    def render_rack_row(self, indent_px, url, name, count, elevation_url):
+        return format_html(
+            """
+            <tr>
+                <td style="padding-left: {}px">
+                    <i class="mdi mdi-folder-open"></i>
+                    <a href="{}">{}</a>
+                </td>
+                <td>{}</td>
+                <td class="text-right noprint">
+                    <a href="{}" class="btn btn-xs btn-primary" title="View elevations">
+                        <i class="mdi mdi-server"></i>
+                    </a>
+                </td>
+            </tr>
+            """,
+            indent_px,
+            url,
+            name,
+            count,
+            elevation_url,
+        )
+
+    def render(self, context):
+        obj = get_obj_from_context(context, self.context_object_key or "object")
+        if not obj:
+            return ""
+
+        rack_groups = context.get("rack_groups", [])
+        stats = context.get("stats", {})
+
+        rows = []
+        for rack_group in rack_groups:
+            rows.append(
+                self.render_rack_row(
+                    getattr(rack_group, "tree_depth", 0) * 8,
+                    rack_group.get_absolute_url(),
+                    str(rack_group),
+                    rack_group.rack_count,
+                    f"{reverse('dcim:rack_elevation_list')}?rack_group={rack_group.pk}",
+                )
+            )
+
+        rows.append(
+            self.render_rack_row(
+                10,
+                "#",
+                "All racks",
+                stats.get("rack_count", 0),
+                f"{reverse('dcim:rack_elevation_list')}?location={obj.pk}",
+            )
+        )
+
+        return format_html(
+            """
+            <div class="panel panel-default">
+                <div class="panel-heading"><strong>Rack Groups</strong></div>
+                <table class="table table-hover panel-body">{}</table>
+            </div>
+            """,
```

Comment:
Can change this method from a fully custom `render()` to just `render_body_content()` and use the `body_wrapper_template_path="components/panel/body_wrapper_generic_table.html"` added in #7825.